File: java/Cube.java
import java.awt.*;

class Cube {
    private static Cube cube;

    private Color[][][] block = new Color[6][3][3];

    private Cube() {
        init();
    }

    static Cube getCube() {
        if (cube == null)
            cube = new Cube();
        return cube;
    }

    void init() {
        for (int z = 0; z < block.length; z++) {
            for (int x = 0; x < block[z].length; x++) {
                for (int y = 0; y < block[z][x].length; y++) {
                    block[z][x][y] = Settings.COLORS[z];
                }
            }
        }
    }

    Color[][][] getBlock() {
        return block;
    }
}

File: c#/Cube.cs
using System.Drawing;

namespace MagicCube
{
    internal class Cube
    {
        private static Cube cube;

        private readonly Color[,,] block = new Color[6, 3, 3];

        private Cube()
        {
            Init();
        }

        internal static Cube GetCube()
        {
            if (cube == null)
                cube = new Cube();
            return cube;
        }

        internal void Init()
        {
            for (int z = 0; z < block.GetLength(0); z++)
            {
                for (int x = 0; x < block.GetLength(1); x++)
                {
                    for (int y = 0; y < block.GetLength(2); y++)
                    {
                        block[z, x, y] = Settings.Colors[z];
                    }
                }
            }
        }

        internal Color[,,] GetBlock()
        {
            return block;
        }
    }
}
